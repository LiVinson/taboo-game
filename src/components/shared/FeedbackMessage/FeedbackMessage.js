import React from "react"
import PropTypes from "prop-types"
import {StyledErrorMessage, StyledSuccessMessage} from "./style"

export const ErrorMessage = ({error, large, light}) => {
    return <StyledErrorMessage light={light} large={large}>{error}</StyledErrorMessage>
}

ErrorMessage.propTypes = {
    error: PropTypes.string,
    large: PropTypes.bool,
    light: PropTypes.bool
}

ErrorMessage.defaultProp = {
    large: false,
    error: "A problem has occured. Please try again."
}

export const SuccessMessage = ({message}) => {
    return <StyledSuccessMessage>{message}</StyledSuccessMessage>
}



SuccessMessage.propTypes = {
    message: PropTypes.string.isRequired,
}
