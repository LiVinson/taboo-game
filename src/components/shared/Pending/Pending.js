import React from 'react'
import PropTypes from 'prop-types'
import { StyledPending, StyledLargePending } from './style'

export class Pending extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			message: this.props.message,
		}
	}

	componentDidMount() {
		const { speed, message } = this.props

		this.setPendingMessage(message, speed)
	}

	setPendingMessage = (message, speed) => {
		this.interval = window.setInterval(() => {
			this.state.message === message + '...'
				? this.setState({ message: message })
				: this.setState(({ message }) => ({ message: message + '.' }))
		}, speed)
	}
	componentDidUpdate(prevProps) {
		if (prevProps.message !== this.props.message) {
			window.clearInterval(this.interval)
			this.setState(
				{
					message: this.props.message,
				},
				this.setPendingMessage(this.props.message, this.props.speed)
			)
		}
	}
	componentWillUnmount() {
		window.clearInterval(this.interval)
	}

	render() {
		const { large } = this.props
		const pending = large ? (
			<StyledLargePending>{this.state.message}</StyledLargePending>
		) : (
			<StyledPending>{this.state.message}</StyledPending>
		)
		return pending
	}
}

Pending.defaultProps = {
	speed: 300,
	large: false,
}

Pending.propTypes = {
	message: PropTypes.string.isRequired,
	speed: PropTypes.number.isRequired,
	large: PropTypes.bool.isRequired,
}
