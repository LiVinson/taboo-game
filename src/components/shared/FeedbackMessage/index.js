import { ErrorMessage, SuccessMessage } from "./FeedbackMessage"

export {ErrorMessage, SuccessMessage}