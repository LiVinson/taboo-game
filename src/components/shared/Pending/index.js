import {Pending} from "./Pending"

export default Pending