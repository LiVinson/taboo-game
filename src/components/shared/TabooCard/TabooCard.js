import React from 'react'
import PropTypes from 'prop-types'
import {
	StyledTabooCard,
	StyledLayeredTabooCard,
	TabooWordContainer,
	TabooBody,
	StyledListItem,
	StyledListItemSecondary,
	StyledListTitle,
} from './style'
import List from 'components/shared/List'
import ButtonContainer from 'components/ButtonContainer'
import Star from 'components/shared/Star'
import {ErrorMessage} from 'components/shared/FeedbackMessage'

//Primary TabooCard component used for main menu and game
export const TabooCard = ({ tabooWord, list, children }) => {
	return (
		<StyledTabooCard>
			<TabooWordContainer>{tabooWord}</TabooWordContainer>
			<TabooBody>{children ? children : <TabooList list={list} />}</TabooBody>
		</StyledTabooCard>
	)
}

TabooCard.propTypes = {
	tabooWord: PropTypes.string.isRequired,
	list: PropTypes.array,
	children: PropTypes.node,
}

//Same as TabooCard, but includes 2 -3 buttons
export const ButtonTabooCard = ({ tabooWord, list, buttons, children, error }) => {
	return (
		<StyledTabooCard>
			<TabooWordContainer>{tabooWord}</TabooWordContainer>
			<TabooBody>
				{children ? children : <TabooList list={list} />}
				{error && <ErrorMessage error={error} />}
			</TabooBody>
			<ButtonContainer buttons={buttons} />
		</StyledTabooCard>
	)
}

ButtonTabooCard.propTypes = {
	tabooWord: PropTypes.string.isRequired,
	buttons: PropTypes.array.isRequired,
	list: PropTypes.array,
	children: PropTypes.node,
}

//Used for background element
export const LayeredTabooCard = ({ tabooWord, list, layer }) => {
	return (
		<StyledLayeredTabooCard layer={layer}>
			<TabooWordContainer>{tabooWord}</TabooWordContainer>
			<TabooBody>
				<TabooList list={list} />
			</TabooBody>
		</StyledLayeredTabooCard>
	)
}

LayeredTabooCard.propTypes = {
	tabooWord: PropTypes.string.isRequired,
	list: PropTypes.array.isRequired,
	layer: PropTypes.string.isRequired,
}

//Used in TabooCard, ButtonTabooCard and FilteredListTabooCard
const TabooList = ({ list, property, specialValue, specialKey, secondary }) => {
	return (
		<List>
			{list.map((listItem, index) => {
				let displayText
				let includeStar = false
				if (typeof listItem === 'object' && listItem !== null && listItem[property]) {
					displayText = listItem[property]

					if (listItem[specialKey] && listItem[specialKey] === specialValue) {
								includeStar = true
					}
				} else {
					displayText = listItem
				}
				return (
					<TabooListItem listItem={displayText} key={index} includeStar={includeStar} secondary={secondary} />
				)
			})}
		</List>
	)
}

TabooList.propTypes = {
	list: PropTypes.array.isRequired,
	property: PropTypes.string,
	specialValue: PropTypes.string,
	specialKey: PropTypes.string,
	secondary: PropTypes.bool,
}

//exported for use in tests
export const TabooListItem = ({ listItem, includeStar, secondary }) => {
	//secondary styling and option to include Star
	return secondary ? (
		<StyledListItemSecondary>
			{listItem}
			{includeStar && <Star />}
		</StyledListItemSecondary>
	) : (
		<StyledListItem>{listItem}</StyledListItem>
	)
}

TabooListItem.propTypes = {
	listItem: PropTypes.any,
	includeStar: PropTypes.bool,
	secondary: PropTypes.bool,
}

export const TabooCardTop = ({ children, margin = false }) => {
	return (
		<StyledTabooCard margin={margin}>
			<TabooWordContainer>{children}</TabooWordContainer>
		</StyledTabooCard>
	)
}

TabooCardTop.propTypes = {
	// children: PropTypes.string.isRequired,
	margin: PropTypes.bool,
}

//ex: playerList array of player objects need to be filtered by team: playerList, team, team1, name, Players, [{}]
export const FilteredTabooList = ({
	unfilteredList,
	listTitle,
	filterKey,
	filterValue,
	displayProperty,
	specialKey,
	specialValue,
	noneMessage,
}) => {

	//filters array based on match of item or item[filterKey] equal to filterValue
	const filteredList = unfilteredList.filter((item) => {
		//If item is an object, filter based on property  value. Otherwise filter based on value of item (i.e. string)
		if (typeof item === 'object' && item !== null) return item[filterKey] === filterValue
		return item === filterValue
	})

	//Displays List of filtered values based on the displayProperty
	return (
		<>
			<StyledListTitle>{listTitle}</StyledListTitle>
			{filteredList.length > 0 ? (
				<TabooList
					list={filteredList}
					property={displayProperty}
					specialKey={specialKey}
					specialValue={specialValue}
					secondary={true}
				/>
			) : (
				<TabooList list={[noneMessage]} secondary={true} />
			)}
		</>
	)
}
