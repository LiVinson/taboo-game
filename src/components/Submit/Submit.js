import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import SubmitCardForm from 'components/SubmitCardForm'
import { ButtonTabooCard } from 'components/shared/TabooCard'
import InstructionsText from 'components/shared/InstructionsText'
import Pending from 'components/shared/Pending'
import { ErrorMessage, SuccessMessage } from 'components/shared/FeedbackMessage'
import { submitCardIdea } from 'store/actions/cardActions'

export class Submit extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			tabooWord: '',
			word1: '',
			word2: '',
			word3: '',
			word4: '',
			word5: '',
			successMsg: '',
		}
	}

	timeOut = null

	handleBackClick = () => {
		this.props.history.push('/home')
	}

	//actions passed from Formik
	handleSubmit = (values, actions) => {
		this.setState(
			{
				tabooWord: values.tabooWord,
				word1: values.word1,
				word2: values.word2,
				word3: values.word3,
				word4: values.word4,
				word5: values.word5,
			},
			() => {
				//Saves card idea in firestore
				const cardIdea = {
					tabooWord: this.state.tabooWord,
					tabooList: [this.state.word1, this.state.word2, this.state.word3, this.state.word4, this.state.word5, ]
				}
				this.props.submitCardIdea(cardIdea).then(() => {
					actions.setSubmitting(false)
					this.displaySuccess()
				})
			}
		)
	}

	componentWillUnmount() {
		clearTimeout(this.timeOut)
	}

	//Removes success message after 2 seconds and resets state, which clears form
	clearSuccess = () => {
		this.timeOut = setTimeout(() => {
			this.setState({
				tabooWord: '',
				word1: '',
				word2: '',
				word3: '',
				word4: '',
				word5: '',
				successMsg: '',
			})
		}, 2000)
	}

	displaySuccess = () => {
		this.setState(
			{
				successMsg: 'Card idea submitted!',
			},
			() => {
				this.clearSuccess()
			}
		)
	}

	render() {
		const buttonInfo = [
			{
				text: 'Back',
				onClick: this.handleBackClick,
			},
			{
				form: 'submitCardForm',
				text: 'Submit',
				type: 'submit',
				disabled: this.props.isPending,
			},
		]
		return (
			<React.Fragment>
				<InstructionsText>
					Submit your taboo card idea and maybe you will see it the next time you play!
				</InstructionsText>
				<ButtonTabooCard tabooWord="Have an Idea?" buttons={buttonInfo}>
					<SubmitCardForm initialValues={this.state} handleSubmit={this.handleSubmit} />
					{this.props.isPending ? <Pending speed={300} message="Submitting card idea." /> : null}
					{this.props.error ? (
						<ErrorMessage error="There was a problem submitting your card idea. Please try again." />
					) : null}
					{this.state.successMsg && <SuccessMessage message={this.state.successMsg} />}
				</ButtonTabooCard>
			</React.Fragment>
		)
	}
}

Submit.propTypes = {
	history: PropTypes.object.isRequired,
	error: PropTypes.string,
	isPending: PropTypes.bool.isRequired,
	submitCardIdea: PropTypes.func.isRequired,
}

const mapStateToProps = (state) => {
	return {
		error: state.cards.error ? state.cards.errorMessage : null ? state.cards.errorMessage : null,
		isPending: state.cards.pending,
	}
}
const mapDispatchToProps = (dispatch) => {
	return {
		submitCardIdea: (cardIdea) => dispatch(submitCardIdea(cardIdea)),
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(Submit)
