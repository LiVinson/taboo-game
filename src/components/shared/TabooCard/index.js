		//If item is an object, filter based on property  value. Otherwise filter based on value of item (i.e. string)
import { TabooCard, ButtonTabooCard, LayeredTabooCard, TabooCardTop, FilteredTabooList} from "./TabooCard"

		//If item is an object, filter based on property  value. Otherwise filter based on value of item (i.e. string)
export { TabooCard, ButtonTabooCard,  LayeredTabooCard, TabooCardTop, FilteredTabooList};

