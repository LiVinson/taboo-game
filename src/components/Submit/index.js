import  Submit from "./Submit"

export default Submit