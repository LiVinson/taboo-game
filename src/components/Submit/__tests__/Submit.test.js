import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'
import theme from '../../../global-design/theme'
import { Submit } from '../Submit'
import { ButtonTabooCard } from 'components/shared/TabooCard'
import SubmitCardForm from 'components/SubmitCardForm'
import Pending from 'components/shared/Pending'
import { ErrorMessage } from 'components/shared/FeedbackMessage'

describe('Submit functionality and rendering', () => {
	const defaultProps = {
		history: {
			push: jest.fn(),
		},
		error: null,
		isPending: false,
		submitCardIdea: jest.fn(() => Promise.resolve()),
	}

	const formValues = {
		tabooWord: 'Will Smith',
		word1: 'Fresh Prince',
		word2: 'Bel-Air',
		word3: 'Jada Pinkett',
		word4: '',
		word5: '',
	}

	const actionsMock = {
		setSubmitting: jest.fn(),
		resetForm: jest.fn(),
	}

	//Used for flashing success message
    beforeEach(() => {
		jest.useFakeTimers()
		
	})

	test('Submit renders without crashing', () => {
		const props = {
			...defaultProps,
		}
		shallow(<Submit {...props} />)
	})

	/* ---------------------SNAPSHOT TESTS & RENDERING ------------------------------- */
	test('Submit renders correctly with default props', () => {
		const props = {
			...defaultProps,
		}
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<Submit {...props} />
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})

	test('Submit renders correctly with isPending props true', () => {
		const props = {
			...defaultProps,
			isPending: true,
		}
		const wrapper = renderer
			.create(
				<ThemeProvider theme={theme}>
					<Submit {...props} />
				</ThemeProvider>
			)
			.toJSON()

		expect(wrapper).toMatchSnapshot()
	})

	test('Submit renders a ButtonTabooCard with a Form', () => {
		const props = {
			...defaultProps,
		}

		const wrapper = shallow(
			<ThemeProvider theme={theme}>
				<Submit {...props} />
			</ThemeProvider>
		)
		//Need to use find to access Submit due to ThemeProvider wrapper and use .dive() for child components due to shallow rendering
		const component = wrapper.find(Submit)
		expect(component.dive().find(ButtonTabooCard).length).toBe(1)
		expect(component.dive().find(SubmitCardForm).length).toBe(1)
	})

	/* --------------------- FUNCTIONALITY & PROPS ------------------------------- */
	test('Submit navigates back to /home when handleBackClick is called', () => {
		const props = {
			...defaultProps,
		}
		const wrapper = shallow(<Submit {...props} />)

		wrapper.instance().handleBackClick()
		expect(props.history.push).toHaveBeenCalled()
		expect(props.history.push).toHaveBeenCalledWith('/home')
	})

	test('Calls props.submitCardIdea with form data on submit', () => {
		const props = {
			...defaultProps,
		}

		const wrapper = shallow(<Submit {...props} />)

		wrapper.instance().handleSubmit(formValues, actionsMock)
		//setTimout gives time for the React.setState to run and it's callback before running test
		setTimeout(() => {
			const state = { ...wrapper.state(), successMsg: '' }

			expect(props.submitCardIdea).toHaveBeenCalled()
			expect(props.submitCardIdea).toHaveBeenCalledWith(state)
			expect(actionsMock.setSubmitting).toHaveBeenCalled()
			expect(actionsMock.resetForm).toHaveBeenCalled()
		}, 75)
	})

	test('Does not render Pending component when props.isPending is false', () => {
		const props = {
			...defaultProps,
		}
		const wrapper = shallow(<Submit {...props} />)
		expect(wrapper.find(Pending)).toHaveLength(0)
	})

	test('renders Pending component when props.isPending is true', () => {
		const props = {
			...defaultProps,
			isPending: true,
		}
		const wrapper = shallow(<Submit {...props} />)
		expect(wrapper.find(Pending)).toHaveLength(1)
	})

	test('Does not render ErrorMessage component when props.isPending is false', () => {
		const props = {
			...defaultProps,
		}
		const wrapper = shallow(<Submit {...props} />)
		expect(wrapper.find(ErrorMessage)).toHaveLength(0)
	})

	test('renders ErrorMessage component when props.isPending is true', () => {
		const props = {
			...defaultProps,
			error: 'There was an error submitting card idea.',
		}
		const wrapper = shallow(<Submit {...props} />)
		expect(wrapper.find(ErrorMessage)).toHaveLength(1)
	})
})
