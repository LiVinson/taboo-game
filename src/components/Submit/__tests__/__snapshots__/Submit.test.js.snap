// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit functionality and rendering Submit renders correctly with default props 1`] = `
Array [
  .c0 {
  font-size: 2rem;
  margin-bottom: 2rem;
  padding: 0 1rem;
  text-align: left;
  color: #333333;
}

.c0:not(:last-child) {
  margin-bottom: 1rem;
}

<p
    className="c0"
  >
    Submit your taboo card idea and maybe you will see it the next time you play!
  </p>,
  .c3 {
  text-align: left;
  font-size: 2.2rem;
  font-family: 'Oswald',sans-serif;
  color: #333333;
}

.c4 {
  margin-bottom: 1.5rem;
  padding-left: 0;
  padding-right: 0;
}

.c7 {
  margin-bottom: 1.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

.c8 {
  font-size: 2rem;
  font-weight: 500;
  text-align: center;
}

.c8::after {
  display: block;
  content: "";
  width: 80%;
  height: 1.5px;
  background-color: #ebebeb;
  margin: 0.8rem auto 1rem auto;
}

.c6 {
  width: 100%;
  padding: 1rem;
  border: none;
  background-color: #ebebeb;
  font-size: 2.2rem;
  border-radius: 2%;
  text-align: center;
  text-transform: uppercase;
}

.c6::-webkit-input-placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6::-moz-placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6:-ms-input-placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6::placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6:focus {
  outline: none;
  border: solid 1px #1982C4;
}

.c5 {
  font-size: 1.6rem;
  display: inline-block;
  font-size: 1.8;
  width: 100%;
  margin-bottom: 0.8rem;
  text-align: center;
  font-weight: 500;
}

.c0 {
  width: 30rem;
  margin: 0 auto;
  padding: 2rem;
  border-radius: 2px;
  position: relative;
  background-color: currentColor;
  color: #1982C4;
}

.c0:before {
  z-index: -1;
  position: absolute;
  content: '';
  bottom: 1.2rem;
  left: 0.5rem;
  width: 50%;
  top: 80%;
  max-width: 300px;
  background: #777;
  box-shadow: 0 1.5rem 1rem rgba(0,0,0,0.4);
  -webkit-transform: rotate(-3deg);
  -ms-transform: rotate(-3deg);
  transform: rotate(-3deg);
}

.c1 {
  background-color: #f7f9f9;
  padding: 1.2rem;
  text-align: center;
  border-radius: 2px;
  font-family: 'Oswald',sans-serif;
  font-size: 4rem;
  color: inherit;
  font-weight: 400;
  text-transform: uppercase;
}

.c2 {
  background-color: #f7f9f9;
  padding: 1.5rem;
  text-align: center;
  border-radius: 2px;
  font-size: 4rem;
  margin-top: 1.5rem;
}

.c10 {
  width: 47%;
  padding: 0.6rem 0rem;
  border: none;
  border-radius: 2px;
  background-color: #f7f9f9;
  font-size: 2.3rem;
  font-family: 'Oswald',sans-serif;
  text-transform: uppercase;
  color: #1982C4;
}

.c10:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c10:active {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c10:focus {
  border: none;
  outline: solid 2px #8AC926;
}

.c10:disabled {
  -webkit-filter: brightness(0.6);
  filter: brightness(0.6);
  cursor: not-allowed;
}

.c11 {
  width: 47%;
  padding: 0.6rem 0rem;
  border: none;
  border-radius: 2px;
  background-color: #f7f9f9;
  font-size: 2.3rem;
  font-family: 'Oswald',sans-serif;
  text-transform: uppercase;
  color: #1982C4;
  color: #f7f9f9;
  background-color: #8AC926;
}

.c11:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c11:active {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c11:focus {
  border: none;
  outline: solid 2px #8AC926;
}

.c11:disabled {
  -webkit-filter: brightness(0.6);
  filter: brightness(0.6);
  cursor: not-allowed;
}

.c11:focus {
  outline: solid 2px #f7f9f9;
}

.c9 {
  margin-top: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div
    className="c0"
  >
    <div
      className="c1"
    >
      Have an Idea?
    </div>
    <div
      className="c2"
    >
      <form
        action="#"
        className="c3"
        id="submitCardForm"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="c4"
        >
          <label
            className="sc-Axmtr c5"
            htmlFor="tabooWord"
          >
            Taboo Word
          </label>
          <input
            className="c6"
            id="tabooWord"
            maxLength={35}
            name="tabooWord"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            tabIndex="1"
            type="text"
            value=""
          />
        </div>
        <div
          className="c7"
        >
          <p
            className="c8"
          >
            Words You Can't Say (Optional)
          </p>
          <label
            className="sc-Axmtr c5"
            htmlFor="word1"
          >
            Word 1
          </label>
          <input
            className="c6"
            id="word1"
            maxLength={35}
            name="word1"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            tabIndex="2"
            type="text"
            value=""
          />
          <label
            className="sc-Axmtr c5"
            htmlFor="word2"
          >
            Word 2
          </label>
          <input
            className="c6"
            id="word2"
            maxLength={35}
            name="word2"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            tabIndex="2"
            type="text"
            value=""
          />
          <label
            className="sc-Axmtr c5"
            htmlFor="word3"
          >
            Word 3
          </label>
          <input
            className="c6"
            id="word3"
            maxLength={35}
            name="word3"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            tabIndex="3"
            type="text"
            value=""
          />
          <label
            className="sc-Axmtr c5"
            htmlFor="word4"
          >
            Word 4
          </label>
          <input
            className="c6"
            id="word4"
            maxLength={35}
            name="word4"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            tabIndex="4"
            type="text"
            value=""
          />
          <label
            className="sc-Axmtr c5"
            htmlFor="word5"
          >
            Word 5
          </label>
          <input
            className="c6"
            id="word5"
            maxLength={35}
            name="word5"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            tabIndex="5"
            type="text"
            value=""
          />
        </div>
      </form>
      
    </div>
    <div
      className="c9"
    >
      <button
        className="c10"
        onClick={[Function]}
      >
        Back
      </button>
      <button
        className="c11"
        disabled={false}
        form="submitCardForm"
        type="submit"
      >
        Submit
      </button>
    </div>
  </div>,
]
`;

exports[`Submit functionality and rendering Submit renders correctly with isPending props true 1`] = `
Array [
  .c0 {
  font-size: 2rem;
  margin-bottom: 2rem;
  padding: 0 1rem;
  text-align: left;
  color: #333333;
}

.c0:not(:last-child) {
  margin-bottom: 1rem;
}

<p
    className="c0"
  >
    Submit your taboo card idea and maybe you will see it the next time you play!
  </p>,
  .c3 {
  text-align: left;
  font-size: 2.2rem;
  font-family: 'Oswald',sans-serif;
  color: #333333;
}

.c4 {
  margin-bottom: 1.5rem;
  padding-left: 0;
  padding-right: 0;
}

.c7 {
  margin-bottom: 1.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

.c8 {
  font-size: 2rem;
  font-weight: 500;
  text-align: center;
}

.c8::after {
  display: block;
  content: "";
  width: 80%;
  height: 1.5px;
  background-color: #ebebeb;
  margin: 0.8rem auto 1rem auto;
}

.c6 {
  width: 100%;
  padding: 1rem;
  border: none;
  background-color: #ebebeb;
  font-size: 2.2rem;
  border-radius: 2%;
  text-align: center;
  text-transform: uppercase;
}

.c6::-webkit-input-placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6::-moz-placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6:-ms-input-placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6::placeholder {
  font-size: 1.8rem;
  font-style: italic;
}

.c6:focus {
  outline: none;
  border: solid 1px #1982C4;
}

.c5 {
  font-size: 1.6rem;
  display: inline-block;
  font-size: 1.8;
  width: 100%;
  margin-bottom: 0.8rem;
  text-align: center;
  font-weight: 500;
}

.c0 {
  width: 30rem;
  margin: 0 auto;
  padding: 2rem;
  border-radius: 2px;
  position: relative;
  background-color: currentColor;
  color: #1982C4;
}

.c0:before {
  z-index: -1;
  position: absolute;
  content: '';
  bottom: 1.2rem;
  left: 0.5rem;
  width: 50%;
  top: 80%;
  max-width: 300px;
  background: #777;
  box-shadow: 0 1.5rem 1rem rgba(0,0,0,0.4);
  -webkit-transform: rotate(-3deg);
  -ms-transform: rotate(-3deg);
  transform: rotate(-3deg);
}

.c1 {
  background-color: #f7f9f9;
  padding: 1.2rem;
  text-align: center;
  border-radius: 2px;
  font-family: 'Oswald',sans-serif;
  font-size: 4rem;
  color: inherit;
  font-weight: 400;
  text-transform: uppercase;
}

.c2 {
  background-color: #f7f9f9;
  padding: 1.5rem;
  text-align: center;
  border-radius: 2px;
  font-size: 4rem;
  margin-top: 1.5rem;
}

.c11 {
  width: 47%;
  padding: 0.6rem 0rem;
  border: none;
  border-radius: 2px;
  background-color: #f7f9f9;
  font-size: 2.3rem;
  font-family: 'Oswald',sans-serif;
  text-transform: uppercase;
  color: #1982C4;
}

.c11:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c11:active {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c11:focus {
  border: none;
  outline: solid 2px #8AC926;
}

.c11:disabled {
  -webkit-filter: brightness(0.6);
  filter: brightness(0.6);
  cursor: not-allowed;
}

.c12 {
  width: 47%;
  padding: 0.6rem 0rem;
  border: none;
  border-radius: 2px;
  background-color: #f7f9f9;
  font-size: 2.3rem;
  font-family: 'Oswald',sans-serif;
  text-transform: uppercase;
  color: #1982C4;
  color: #f7f9f9;
  background-color: #8AC926;
}

.c12:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c12:active {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c12:focus {
  border: none;
  outline: solid 2px #8AC926;
}

.c12:disabled {
  -webkit-filter: brightness(0.6);
  filter: brightness(0.6);
  cursor: not-allowed;
}

.c12:focus {
  outline: solid 2px #f7f9f9;
}

.c10 {
  margin-top: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c9 {
  font-size: 1.5rem;
  font-style: italic;
  color: #1982C4;
}

<div
    className="c0"
  >
    <div
      className="c1"
    >
      Have an Idea?
    </div>
    <div
      className="c2"
    >
      <form
        action="#"
        className="c3"
        id="submitCardForm"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="c4"
        >
          <label
            className="sc-Axmtr c5"
            htmlFor="tabooWord"
          >
            Taboo Word
          </label>
          <input
            className="c6"
            id="tabooWord"
            maxLength={35}
            name="tabooWord"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            tabIndex="1"
            type="text"
            value=""
          />
        </div>
        <div
          className="c7"
        >
          <p
            className="c8"
          >
            Words You Can't Say (Optional)
          </p>
          <label
            className="sc-Axmtr c5"
            htmlFor="word1"
          >
            Word 1
          </label>
          <input
            className="c6"
            id="word1"
            maxLength={35}
            name="word1"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            tabIndex="2"
            type="text"
            value=""
          />
          <label
            className="sc-Axmtr c5"
            htmlFor="word2"
          >
            Word 2
          </label>
          <input
            className="c6"
            id="word2"
            maxLength={35}
            name="word2"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            tabIndex="2"
            type="text"
            value=""
          />
          <label
            className="sc-Axmtr c5"
            htmlFor="word3"
          >
            Word 3
          </label>
          <input
            className="c6"
            id="word3"
            maxLength={35}
            name="word3"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            tabIndex="3"
            type="text"
            value=""
          />
          <label
            className="sc-Axmtr c5"
            htmlFor="word4"
          >
            Word 4
          </label>
          <input
            className="c6"
            id="word4"
            maxLength={35}
            name="word4"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            tabIndex="4"
            type="text"
            value=""
          />
          <label
            className="sc-Axmtr c5"
            htmlFor="word5"
          >
            Word 5
          </label>
          <input
            className="c6"
            id="word5"
            maxLength={35}
            name="word5"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            tabIndex="5"
            type="text"
            value=""
          />
        </div>
      </form>
      <p
        className="c9"
      >
        Submitting card idea.
      </p>
      
    </div>
    <div
      className="c10"
    >
      <button
        className="c11"
        onClick={[Function]}
      >
        Back
      </button>
      <button
        className="c12"
        disabled={true}
        form="submitCardForm"
        type="submit"
      >
        Submit
      </button>
    </div>
  </div>,
]
`;
